methylDackel.wdl

```diff
@@ -29,75 +31,169 @@ workflow methylDackel {
 
     GenomeResources ref = resources[reference]
 
-    call methylDackel_extract {
+    call methylDackelExtract {
         input:
             bam = bam,
             bai = bai,
-            prefix = outputFileNamePrefix,
+            outputFileNamePrefix = outputFileNamePrefix,
             fasta = ref.fasta,
             modules = "methyldackel/0.6.1 ~{ref.genomeModule}"
     }
 
+    if ( doMbias ){
+        call extractChromosomes{
+            input:
+            bam = bam
+        }
+
+        scatter ( chr in extractChromosomes.chromosomes ) {
+            call methylDackelMbias {
+                input:
+                    bam = bam,
+                    bai = bai,
+                    chr = chr,
+                    fasta = ref.fasta,
+                    outputFileNamePrefix = outputFileNamePrefix,
+                    modules = "methyldackel/0.6.1 samtools/1.16.1 ~{ref.genomeModule}"
+            }
+        }      
+    
+        Array[File?] mbiasTsvs = methylDackelMbias.mbias_tsv
+        Array[File?] mbiasSvgs = methylDackelMbias.mbias_svg_files
+
+        call concatMbiasFiles {
+            input:
+                inputTsvs = mbiasTsvs,
+                inputSvgs = mbiasSvgs,
+                outputFileNamePrefix = outputFileNamePrefix
+        }
+    }
 
     meta {
         author: "Gavin Peng"
         email: "gpeng@oicr.on.ca"
-        description: "Workflow to run bwa-meth, the fast aligner for EM-seq/BS-Seq reads. Prior to alignment, adatper trimming and quality filtering are performed. Readgroup information to be injected into the bam header needs to be provided.  The workflow can also split the input data into a requested number of chunks, align each separately then merge the separate alignments into a single bam file.  This decreases the workflow run time. Final bam file also applied markDuplicates."
+        description: "Workflow to run methylDackel, will process a coordinate-sorted and indexed BAM or CRAM file containing some form of BS-seq or EM-seq alignments and extract per-base methylation metrics from them. The extract task generates bedGraph files, by default generates only CpG metrics, option can be set to also generate CHH and CHG metrics. Mbias task generates tsv file for methylation bias metrics and a svg graph for visualizing mbias (only for chromosome 1 here)."
         dependencies: [
          {
             name: "methyldackel/0.6.1",
             url: "https://github.com/dpryan79/MethylDackel"
                         }
         ]
         output_meta: {
-        extract_bedgraph: {
-            description: "bedGraph output from methyDackel extract",
-            vidarr_label: "extract_bedgraph"
-        }
+            extract_bedgraph: {
```

Comment:
output_meta with file labels does not correspond to output

there are 3 bedgraph files generated, only one is being indicated here and the name does not match any of the output labels

Reply:
Thanks! fixed.

Reply:
this doesn't appear to be fixed

shouldn't the output Meta have entries for
extract_CpGbedgraph 
extract_CHGbedgraph
extract_CHHbedgraph

i see only
extract_bedgraph
which is not an output label

Reply:
if you look at this commit: https://github.com/oicr-gsi/methylDackel/pull/1/commits/c0309f7ae480f9149cb9fd02e412d7607dc2e457
the requried change already fixed.

Reply:
ok, sorry, i think i was looking at this incorrectly and didn't see the change.  I see it now